dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -108,6 +108,7 @@ public class GitHubSourceCodeRepo extends SourceCodeRepoInterface {
 
     public static final String GITHUB_ABUSE_LIMIT_REACHED = "GitHub abuse limit reached";
     public static final int GITHUB_MAX_CACHE_AGE_SECONDS = 30; // GitHub's default max-cache age is 60 seconds
+    public static final Pattern GIT_BRANCH_TAG_PATTERN = Pattern.compile("^refs/(tags|heads)/((?!.*//)(?!.*\\\\)(?!.*@)(?!.*\\[)(?!.*\\?)(?!.*~)(?!.*\\.\\.)[\\p{Punct}\\p{L}\\d\\-_/]+)$");
```

Comment:
Similar feelings about this as I had about unusual characters in `SourceFile` paths.  Do we really want stuff like semicolons and pipes in our version names?  FWIW, see the "Naming branches and tags" section at the end of this GitHub "Getting Started" guide page:
https://docs.github.com/en/get-started/using-git/dealing-with-special-characters-in-branch-and-tag-names

Reply:
I don't know, I think it's worth a discussion. 

The weird part is that if I understand that section of code, the regex wasn't even really being used as a guard. Rather, we're using it to capture the type of git reference and then mutate it to a different form, the fact that it disallows certain combinations of characters was almost a side-effect. 

So I relaxed it to include thing's like @aofarrel 's name, internal scripts, but disallowed the various things that git disallows. FWIW, I'm not even sure many of these can make it into github in the first place!
That said, I think it would be nice to standardise on something so that I can re-use it for bitbucket, hosted workflows, etc. 

Will poke around and see if GitHub is more strict than git in some ways since that may simplify the discussion

Reply:
> The new regexp allows `^` and `:`, not sure if that's ok or not, they're listed as disallowed in https://git-scm.com/docs/git-check-ref-format

FWIW, these both get stripped by github when creating branches (at least from the UI)